servant-client/test/Servant/ClientSpec.hs

```diff
@@ -296,6 +299,22 @@ sucessSpec = beforeAll (startWaiApp server) $ afterAll endWaiApp $ do
             return $
               result === Right (cap, num, flag, body)
 
+    describe "staticClient" $ do
+      it "allows to pass in the BaseUrl to staticClient" $ \(_, baseUrl) -> do
+        let
+            (getGet'  :: C.Manager -> ClientM Person)
+              :<|> (_ ::  C.Manager -> SCR.ClientM NoContent)
+              :<|> (_ ::  String -> C.Manager ->  SCR.ClientM Person)
+              :<|> (_ ::  Person -> C.Manager ->  SCR.ClientM Person)
+              :<|> (_ ::  Maybe String -> C.Manager ->  SCR.ClientM Person)
+              :<|> (_ ::  [String] -> C.Manager ->  SCR.ClientM [Person])
+              :<|> (_ ::  Bool -> C.Manager ->  SCR.ClientM Bool)
+              :<|> (_ ::  HTTP.Method -> C.Manager -> SCR.ClientM (Int, BS.ByteString, MediaType, [HTTP.Header], C.Response BS.ByteString))
+              :<|> (_ ::  HTTP.Method -> C.Manager -> SCR.ClientM (Int, BS.ByteString, MediaType, [HTTP.Header], C.Response BS.ByteString))
+              :<|> (_ ::  String -> Maybe Int -> Bool -> [(String, [Rational])] -> C.Manager -> SCR.ClientM (String, Maybe Int, Bool, [(String, [Rational])]))
+              :<|> (_ ::  C.Manager -> SCR.ClientM (Headers TestHeaders Bool))
+              :<|> (_ ::  C.Manager -> SCR.ClientM NoContent) = staticClient baseUrl api
+        (left show <$> runExceptT (getGet' manager)) `shouldReturn` Right alice
```

Comment:
In theory that's not the case though. Is it the case here because `Enter` is missing a fundep? (Which may not be fixable though - there were some issues with a sensible fundep on GHC 7.6).